### DISCUSSION TITLE
NekRS only supports GNU compilers

### DISCUSSION THREAD

**Discussion by itgreenquist** [Nek5000/nekRS#412] [Compiling]
Hi,

I'm trying to install nekRS for the first time using a Mac. I followed the instructions to download the code from Github, but whenever I run the nrsconfig script I get the following error.

-- Detecting Fortran/C Interface
-- Detecting Fortran/C Interface - Found GLOBAL and MODULE mangling
CMake Error at CMakeLists.txt:64 (message):
  NekRS only supports GNU compilers
Call Stack (most recent call first):
  CMakeLists.txt:74 (check_compiler_id)

I'm using Cmake version 3.21.3.

Any idea what I can do to fix this error and finish installing the code?

Thanks,
Ian

**Comment by stgeke** [comment]
Do you have the GNU compilers installed on your MAC?

**Reply by itgreenquist** [reply]
I have gcc installed. Are there other compilers I need?

**Reply by stgeke** [reply]
Yes `gfortran` is required as well. Moreover, please ensure that the MPI wrappers point to the correct GNU compilers.

**Reply by itgreenquist** [reply]
I checked and I have gfortran installed. How do I check the MPI wrappers?

**Reply by stgeke** [reply]
mpicc --show
mpic++ --show
mpif77 --show

**Reply by itgreenquist** [reply]
(moose) ~ % mpicc -show
x86_64-apple-darwin13.4.0-clang -march=core2 -mtune=haswell -I/Users/6ig/miniconda3/envs/moose/include -L/Users/6ig/miniconda3/envs/moose/lib -Wl,-rpath,/Users/6ig/miniconda3/envs/moose/lib -framework OpenCL -Wl,-commons,use_dylibs -I/Users/6ig/miniconda3/envs/moose/include -L/Users/6ig/miniconda3/envs/moose/lib -lmpi -lpmpi

(moose) ~ % mpic++ -show
x86_64-apple-darwin13.4.0-clang++ -march=core2 -mtune=haswell -I/Users/6ig/miniconda3/envs/moose/include -L/Users/6ig/miniconda3/envs/moose/lib -Wl,-rpath,/Users/6ig/miniconda3/envs/moose/lib -framework OpenCL -Wl,-commons,use_dylibs -I/Users/6ig/miniconda3/envs/moose/include -L/Users/6ig/miniconda3/envs/moose/lib -lmpicxx -lmpi -lpmpi

(moose) ~ % mpif77 -show
x86_64-apple-darwin13.4.0-gfortran -march=core2 -mtune=haswell -I/Users/6ig/miniconda3/envs/moose/include -L/Users/6ig/miniconda3/envs/moose/lib -Wl,-rpath,/Users/6ig/miniconda3/envs/moose/lib -framework OpenCL -Wl,-commons,use_dylibs -I/Users/6ig/miniconda3/envs/moose/include -I/Users/6ig/miniconda3/envs/moose/include -L/Users/6ig/miniconda3/envs/moose/lib -lmpifort -lmpi -lpmpi

**Reply by stgeke** [reply]
Looks like mpicc and mpic++ are pointing to clang that's why you get the error message.

**Reply by itgreenquist** [reply]
I had to go back and rebuild packages, but I got mpi to point to GNU compilers:

(moose) ~ % which gcc
/usr/bin/gcc
(moose) ~ % which g++
/usr/bin/g++
(moose) ~ % which gfortran
/Users/6ig/miniconda3/envs/moose/bin/gfortran
(moose) ~ % mpicc -show
gcc -I/Users/6ig/miniconda3/envs/moose/include -L/Users/6ig/miniconda3/envs/moose/lib -lmpi
(moose) ~ % mpicxx -show
g++ -I/Users/6ig/miniconda3/envs/moose/include -L/Users/6ig/miniconda3/envs/moose/lib -lmpi
(moose) ~ % mpif77 -show
gfortran -I/Users/6ig/miniconda3/envs/moose/include -Wl,-flat_namespace -Wl,-commons,use_dylibs -I/Users/6ig/miniconda3/envs/moose/lib -L/Users/6ig/miniconda3/envs/moose/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi

However, I'm still getting the same error. I probably messed something up while rebuilding MPI. I'd appreciate any ideas or suggestions of what's going wrong here.

(moose) contrib/nekRS % ./nrsconfig 
-- Detecting Fortran/C Interface
CMake Error: Generator: execution of make failed. Make command was: /Users/6ig/miniconda3/envs/mpi-test/bin/make -f Makefile FortranCInterface && 
Failed to compile
CMake Error at CMakeLists.txt:64 (message):
  NekRS only supports GNU compilers
Call Stack (most recent call first):
  CMakeLists.txt:74 (check_compiler_id)


-- Configuring incomplete, errors occurred!
See also "/Users/6ig/projects/cardinal/contrib/nekRS/build/CMakeFiles/CMakeOutput.log".
See also "/Users/6ig/projects/cardinal/contrib/nekRS/build/CMakeFiles/CMakeError.log".
